### PR TITLE
Started with integration tests

### PR DESCRIPTION
**Pull request by maltesander**
- Create a cluster
- Scale a cluster up and down
- Check if container ports and monitoring is set up correctly

fixes: [36](https://github.com/stackabletech/issues/issues/36)

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          profile: minimal
+          toolchain: stable
+          components: rustfmt
+          override: true
+      - uses: actions-rs/cargo@v1.0.3
+        with:
+          command: fmt
+          args: --all -- --check
+
+  clippy:
```
Comment (lfrancke) [inline .github/workflows/rust.yml:49]: Could you look at e.g. operator-rs and include rustdoc here?

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      - name: Run clippy manually without annotations
+        if: ${{ !steps.check_permissions.outputs.has-permission }}
+        run: cargo clippy --all-targets -- -D warnings
+
+  security_audit:
+    name: Run security audit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2.3.4
+      - uses: actions-rs/audit-check@v1.2.0
```
Comment (lfrancke) [inline .github/workflows/rust.yml:83]:
```suggestion
      - uses: actions-rs/audit-check@v1.2.6
```

File: Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+[package]
+authors = ["Malte Sander <malte.sander.it@gmail.com>"]
+description = "Integration tests for the Stackable Operator for Apache Kafka"
+edition = "2018"
+license = "OSL-3.0"
+name = "kafka-operator-integration-tests"
```
Comment (lfrancke) [inline Cargo.toml:6]: Could you include a single line here

`repository = "https://github.com/stackabletech/kafka-operator-integration-tests"`

File: Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+version = "0.1.0-nightly"
```
Comment (lfrancke) [inline Cargo.toml:7]:
```suggestion
version = "0.1.0"
```

I talked to @siegfriedweber  and we decided to skip the nightly tag for integration tests

**Review by lfrancke — CHANGES_REQUESTED** [review]

File: .github/workflows/rust.yml (hunk 2 above)
Comment (maltesander) [reply .github/workflows/rust.yml]: There is no 1.2.6? I used it without checking and did not work. Ill go back to [1.2.0](https://github.com/actions-rs/audit-check/releases).

**Review by razvan — APPROVED** [review]
lgtm

**Comment by backstreetkiwi** [comment]
Hi @maltesander, I tested this with Jenkins and it failed. Maybe I didn't fulfill all prerequisites?

Logs: https://ci.stackable.tech/job/Kafka%20Operator%20Integration%20Tests/1/artifact/target/test_output.log
Cluster setup: https://ci.stackable.tech/job/Kafka%20Operator%20Integration%20Tests/1/artifact/target/stackable-versions.txt

**Comment by backstreetkiwi** [comment]
UPDATE: I included a ZooKeeper service to the test cluster definition, now it works: https://ci.stackable.tech/job/Kafka%20Operator%20Integration%20Tests/3/

**Review by backstreetkiwi — APPROVED** [review]
Black-box-test successful: https://ci.stackable.tech/job/Kafka%20Operator%20Integration%20Tests/3/

File: Cargo.toml [hunk 5]
```diff
@@ -4,11 +4,12 @@ description = "Integration tests for the Stackable Operator for Apache Kafka"
...
 license = "OSL-3.0"
 name = "kafka-operator-integration-tests"
-version = "0.1.0-nightly"
+repository = "https://github.com/stackabletech/kafka-operator-integration-tests"
+version = "0.1.0"
 
 [dependencies]
 anyhow = "1.0"
-integration-test-commons = { git = "https://github.com/stackabletech/integration-test-commons.git", branch = "remove_conditions_trait" }
+integration-test-commons = { git = "https://github.com/stackabletech/integration-test-commons.git", branch = "main" }
```
Comment (lfrancke) [inline Cargo.toml:12]: Can we move to a released tag here?

**Review by lfrancke — APPROVED** [review]
Only a single comment but looks good and can be merged like this

**Review by lfrancke — APPROVED** [review]

**Pull request merged**